artemis-protocols/artemis-openwire-protocol/src/main/java/org/apache/activemq/artemis/core/protocol/openwire/OpenWireConnection.java

```diff
@@ -641,13 +647,37 @@ public void fail(ActiveMQException me, String message) {
          }
       }
       try {
-         protocolManager.removeConnection(this.getConnectionInfo(), me);
+         if (this.getConnectionInfo() != null) {
+            protocolManager.removeConnection(this.getConnectionInfo(), me);
+         }
       } catch (InvalidClientIDException e) {
          ActiveMQServerLogger.LOGGER.warn("Couldn't close connection because invalid clientID", e);
       }
       shutdown(true);
    }
 
+   private void delayedStop(final int waitTime, final String reason, Throwable cause) {
+      if (waitTime > 0) {
+         try {
+            server.getExecutorFactory().getExecutor().execute(new Runnable() {
+               @Override
+               public void run() {
+                  try {
+                     Thread.sleep(waitTime);
```

Comment:
Sleeping here seems like it might have a ripple on the other resources using the executor, maybe use a scheduled executor from the pool in ActiveMQServer ?

Reply:
Agreed, I will make that change and use a scheduled executor instead.

Reply:
I won't have time today but I will fix my patch and push it up first thing monday morning.